cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -848,4 +884,99 @@ private void maybeReexecuteTasks() {
       }
     }
   }
+
+  public class ExecutionNotifierPayload {
+    private String _startedBy;
+    private String _endedBy;
+    private long _startMs;
+    private long _endMs;
+    private UserTaskManager.UserTaskInfo _userTaskInfo;
+    private boolean _executionSucceeded;
+    private String _operation;
+    private boolean _startedByUser;
+    private Throwable _exception = null;
+    private String _actionUuid = null;
+
+    public ExecutionNotifierPayload() {
+      _startMs = _executionStartMs;
+      _endMs = _time.milliseconds();
+      _userTaskInfo = null;
+
+      _startedBy = "UNKNOWN";
+      _operation = "UNKNOWN";
+      _startedByUser = false;
+      if (_userTaskManager != null && _uuid != null) {
+        _userTaskInfo = _userTaskManager.getUserTaskById(_uuid);
+        _actionUuid = _uuid;
+        if (_userTaskInfo != null) {
```

Comment:
You are right about `_userTaskInfo != null`. I think it makes sense to store `UserTaskManager` in executor, the reason being since `Executor` is executing some proposal based on UserTasks, it makes sense for it to be able to lookup the context of the request (contained in UserTaskManager) based on UUID. Passing some additional information also involves expanding existing signature of methods to include parameters,  like `Endpoint`, just to pass the information to `Executor`.

Reply:
yes, so we need to first work on [598](https://github.com/linkedin/cruise-control/issues/598)... otherwise we are introducing bug here.

Reply:
Stored the UserTaskInfo at the start of proposal execution, so the issue is fixed.

Reply:
I am really confused -- @alexwang08 @kidkun Am I looking at an outdated patch? Let's address these issues before we approve this patch.

Reply:
@efeg Alex address the issue by `Stored the UserTaskInfo at the start of proposal execution` and that operation is synchronized, he can get the information he needs.

Reply:
@efeg I made updates after Kun's comments, hence this snippet is outdated